cpp/ql/lib/semmle/code/cpp/commons/NullTermination.qll

```diff
@@ -3,11 +3,14 @@ private import semmle.code.cpp.models.interfaces.ArrayFunction
 private import semmle.code.cpp.models.implementations.Strcat
 import semmle.code.cpp.dataflow.DataFlow
 
-private predicate mayAddNullTerminatorHelper(Expr e, VariableAccess va, Expr e0) {
-  exists(StackVariable v0, Expr val |
-    exprDefinition(v0, e, val) and
-    val.getAChild*() = va and
-    mayAddNullTerminator(e0, v0.getAnAccess())
+/**
+ * Holds if the expression `e` assigns something including `va` to a
+ * stack variable `v0` that is later null terminated at `e0`.
```

Comment:
```suggestion
 * Holds if the expression `e` assigns something including `va` to a
 * stack variable `v0`.
```
`e0` isn't a parameter anymore, and I guess the "later null terminated"-part is now moved to `mayAddNullTerminator`, right?

Reply:
Good point, well spotted.